Dockerfile

```diff
@@ -1,19 +1,45 @@
 # syntax=docker/dockerfile:1
 
+# ================
+# CONFIGURATION
+# ================
+# ShellCheck version
 ARG VERSION
 
-FROM --platform=darwin/arm64 ghcr.io/homebrew/core/shellcheck:${VERSION} AS darwin-arm64
+# ================
+# DARWIN ARM64
+# ================
+FROM --platform=darwin/arm64 ghcr.io/homebrew/core/shellcheck:$VERSION AS darwin-arm64
 
+# ================
+# ARCHIVES
+# ================
 FROM alpine AS archives
+ARG VERSION
 
-RUN apk add --no-cache bash curl tar unzip xz
+# Install packages
+RUN apk add --no-cache \
+  findutils \
+  tar \
+  unzip \
+  xz
 
-COPY scripts/download_and_archive.sh /scripts/download_and_archive.sh
+# Copy Darwin arm64 binary
+COPY --from=darwin-arm64 shellcheck/$VERSION/bin/shellcheck /shellcheck.darwin.arm64.data/
 
-ARG VERSION
-COPY --from=darwin-arm64 shellcheck/${VERSION}/bin/shellcheck /shellcheck.darwin.arm64.data/
+# Copy scripts directory
+COPY scripts /scripts
+
+# Modify scripts permissions
+RUN find /scripts -type f -iname "*.sh" -exec chmod +x "{}" \;
```

Comment:
this should not be needed. Make sure to add execution permission with git add --chmod=+x if required.

However, __commons.sh should NOT have execution permission, as it's not meant to be executed.

Reply:
Removed :)